### PR TITLE
test: add comprehensive unit tests for ring module

### PR DESCRIPTION
**Pull request by sanity**
## Problem

The ring module had low test coverage in several key files:
- `connection_manager.rs`: 22.1%
- `peer_key_location.rs`: 18.0%
- `location.rs`: 64.0%
- `score.rs`: 0% (not even included in mod.rs)

Additionally, the `Score` type's `Ord` implementation had a potential bug where `partial_cmp().unwrap_or(Equal)` would treat NaN values as equal to everything, violating `Ord` requirements.

### Bug Fix
- Fixed `Score::Ord` to use `total_cmp()` for consistent NaN handling. The previous implementation would cause undefined behavior in sorting algorithms when NaN values were present.

### New Tests (~100 tests added)

**score.rs** (8 tests):
- Normal value ordering, negative values, NaN handling, infinity handling
- Sorting with mixed values (verifies NaN sorts consistently)

**location.rs** (~35 tests):
- Distance calculations (wrapping at 0.5, normalization)
- Location arithmetic (add distance, boundary conditions)
- IPv4/IPv6 address mapping (Sybil mitigation via subnet masking)
- Hash consistency for HashMap/HashSet usage
- Contract key location computation

**peer_key_location.rs** (~25 tests):
- PeerAddr variants (Unknown/Known)
- Ordering (unknown addresses sort last)
- Display formatting
- IPv6 support
- Serialization

**connection_manager.rs** (~30 tests):
- Initial state and configuration (gateway vs regular mode)
- `should_accept` logic (min/max connections, already connected peers)
- Transient connection budget enforcement
- Routing with skip lists
- Peer identity updates

## Test Plan
- [x] All 148 ring module tests pass
- [x] Clippy passes with no warnings
- [x] Code formatted

[AI-assisted - Claude]

### CODE REVIEW

File: crates/core/src/ring/score.rs [hunk 1]
```diff
@@ -1,8 +1,28 @@
...
+#![allow(dead_code)] // Score will be used as refactoring progresses
+
+/// A score value used for ranking peers and contracts.
+///
+/// Uses `f64::total_cmp` for ordering to ensure consistent behavior with NaN values.
+/// This is important because the previous implementation would treat NaN as equal
+/// to everything, violating `Ord` requirements and potentially causing subtle bugs
+/// in sorting algorithms.
+#[derive(PartialEq, Clone, Copy, Debug)]
 pub struct Score(pub f64);
```
Comment (iduartgomez) [inline crates/core/src/ring/score.rs]: @claude wan'0t this in use already? why is necessry to mark this module as allow(dead_code) now?
